### PR TITLE
WIP: Playground redesign

### PR DESCRIPTION
**Pull request by JuroUhlar**
JIRA: https://fingerprintjs.atlassian.net/browse/INTER-510
Demo: https://staging.fingerprinthub.com/playground

* Redesigned the Playground according to the original [Figma design](https://www.figma.com/design/Rw0BBJ46Fpbx5s5lWflhVy/Fingerprint-Hub-Demo-Project?node-id=491-68477&m=dev) unifying the esthetic with the main website 
* Added *How to use this demo section* featuring a new `Accordion` component
* Added a *Learn more* section and made the tiles switch to horizontal scrolling behavior on mobile devices
* Signal tables are collapsible now using a new `Collapsible` component
* Improved latency perception by showing the visitor ID [right away while waiting for the Server API result](https://fingerprintjs.atlassian.net/browse/INTER-644)
* Improved JSON viewing experience: 
  * Researched and found a JSON viewer library with (hopefully) sufficient customization options
  * Copy button for the entire JSON and individual objects within
  * JSON nodes can collapse and expand on click
  * Better (more precise) syntax highlighting
  * The new library required simplifying some of the e2e tests but I think they still work well enough
  * Unified scrollbar appearance across browsers using a new `ScrollArea` component
* Moved Playground to the `app` router
* Smarter use of `accuracyRadius` - if large, we only show the user their country, and the map is zoomed out
* Refactored `FancyNumberedList` into a reusable component
* Refactored `ResourceLinks` into a reusable component
* Fixed React mark-up errors in SVG components
* CI: Fixed E2E `report-status` job condition

### CODE REVIEW

File: src/client/components/common/componentUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  Component: T,
+  className: string,
+  displayName: string,
+) {
+  type ComponentRef = React.ElementRef<T>;
+  type ComponentProps = React.ComponentPropsWithoutRef<T & React.RefAttributes<T>>;
+
+  const WrappedComponent = React.forwardRef<ComponentRef, ComponentProps>(
+    ({ className: incomingClassName, ...props }, ref) => (
+      // @ts-ignore Don't know how to fix this
```
Comment (JuroUhlar) [inline src/client/components/common/componentUtils.tsx:18]: @TheUnderScorer @ilfa if you know how to fix this TS error, I would be in your debt

**Review by ilfa — APPROVED** [review]

File: e2e/playground.spec.ts [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
 
+function parseRequestId(inputString: string) {
+  const regex = /requestId:\s*"([^"]+)"/;
+  const match = inputString.match(regex);
+
+  if (match && match[1]) {
+    return match[1];
+  } else {
+    return null;
+  }
```
Comment (TheUnderScorer) [inline e2e/playground.spec.ts:27]: nit: let's unwrap `else` here

File: e2e/playground.spec.ts [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+    (await (await page.getByTestId(TEST_ID.agentResponseJSON)).textContent()) ?? 'Agent response not found';
+  return agentResponse;
 };
 
 const getServerResponse = async (page: Page) => {
   const serverResponse =
-    (await (await page.getByTestId(PLAYGROUND_TAG.serverResponseJSON)).textContent()) ?? 'Server response not found';
-  return JSON.parse(serverResponse);
+    (await (await page.getByTestId(TEST_ID.serverResponseJSON)).textContent()) ?? 'Server response not found';
+  return serverResponse;
```
Comment (TheUnderScorer) [inline e2e/playground.spec.ts:16]: These nested awaits make it less readable, pls try to simplify this

File: src/client/components/common/CodeSnippet/CodeSnippet.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    padding: '16px',
+    fontFamily: 'JetBrains Mono, monospace',
+    fontSize: '16px',
+    overflow: 'auto',
+    marginTop: '0',
+    marginBottom: '0',
+    marginLeft: '0',
+    marginRight: '0',
+  };
+  const PRISM_CODE_TAG_PROPS = { style: { color: '#c92c2c', font: 'inherit' as const } };
```
Comment (TheUnderScorer) [inline src/client/components/common/CodeSnippet/CodeSnippet.tsx:45]: nit: Let's move this constants outside of the component

File: src/client/components/common/componentUtils.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      <Component ref={ref} className={classnames(className, incomingClassName)} {...props} />
+    ),
+  );
+
+  WrappedComponent.displayName = displayName;
+  return WrappedComponent;
+}
```
Comment (TheUnderScorer) [inline src/client/components/common/componentUtils.tsx:25]: Try this:

```suggestion
import classnames from 'classnames';
import React, { ComponentType, RefAttributes } from 'react';

/**
 * Reusable function if you just need to extend an unstyled primitive (like a Radix component) with a classname
 * https://www.radix-ui.com/primitives/docs/guides/styling#extending-a-primitive
 */
export function extendUnstyledPrimitiveWithClass<P extends { className?: string }, R>(
  Component: ComponentType<P & RefAttributes<R>>,
  className: string,
  displayName: string,
) {
  const WrappedComponent = React.forwardRef<R, P>(({ className: incomingClassName, ...props }, ref) => (
    <Component ref={ref} className={classnames(className, incomingClassName)} {...(props as P)} />
  ));

  WrappedComponent.displayName = displayName;
  return WrappedComponent;
}

```

File: e2e/playground.spec.ts (hunk 2 above)
Comment (JuroUhlar) [reply e2e/playground.spec.ts]: Thanks, fixed: 92cd7afa8ed5538543133f02f780c9bd90e0686e

File: e2e/playground.spec.ts (hunk 3 above)
Comment (JuroUhlar) [reply e2e/playground.spec.ts]: Thanks, fixed: https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/commit/92cd7afa8ed5538543133f02f780c9bd90e0686e

File: src/client/components/common/CodeSnippet/CodeSnippet.tsx (hunk 4 above)
Comment (JuroUhlar) [reply src/client/components/common/CodeSnippet/CodeSnippet.tsx]: Thanks, fixed: https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/commit/92cd7afa8ed5538543133f02f780c9bd90e0686e

File: src/client/components/common/componentUtils.tsx (hunk 5 above)
Comment (JuroUhlar) [reply src/client/components/common/componentUtils.tsx]: Genius! Thank you <3

**Review by TheUnderScorer — DISMISSED** [review]

**Review by ilfa — DISMISSED** [review]

**Review by ilfa — APPROVED** [review]

**Pull request merged**